linked_list/linked_list.py

```diff
@@ -86,11 +175,41 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def reverse(self):

```

Comment:
✨ Very creative solution! If you wanted to refactor, think about how you might be able to do this with just one traversal of the list (one while loop). Hint: Pointers will play a big part!